autotx/AutoTx.py

```diff
@@ -1,73 +1,119 @@
 
-    def run(self, prompt: str, non_interactive: bool):
-        print(f"Defining goal for prompt: '{prompt}'")
-       
-        agents_information = self.get_agents_information()
+    def run(self, prompt: str, non_interactive: bool, silent: bool = False):
+        print("Running AutoTx with the following prompt: ", prompt)
+
+        user_proxy = UserProxyAgent(
+            name="user_proxy",
+            is_termination_msg=lambda x: x.get("content", "") and x.get("content", "").rstrip().endswith("TERMINATE"),
+            human_input_mode="NEVER",
+            max_consecutive_auto_reply=20,
+            system_message=f"You are a user proxy. You will be interacting with the agents to accomplish the tasks.",
+            llm_config=self.get_llm_config(),
+            code_execution_config=False,
+        )
+
+        agents_information = self.get_agents_information(self.agents)
 
         goal = build_goal(prompt, agents_information, self.manager.address, non_interactive)
 
-        print(f"Defining tasks for goal: '{goal}'")
-        tasks: list[Task] = define_tasks(goal, agents_information, self.agents)
-   
-        self.run_for_tasks(tasks, non_interactive)
-
-    def run_for_tasks(self, tasks: list[Task], non_interactive: bool):
-        print(f"Running tasks...")
-        Crew(
-            agents=self.agents,
-            tasks=tasks,
-            verbose=self.config.verbose,
-            process=Process.sequential,
-            function_calling_llm=open_ai_llm,
-        ).kickoff()
-
-        self.manager.send_tx_batch(self.transactions, require_approval=not non_interactive)
+        verifier_agent = AssistantAgent(
+            name="verifier",
+            is_termination_msg=lambda x: x.get("content", "") and x.get("content", "").rstrip().endswith("TERMINATE"),
+            system_message=dedent(
+                    """
+                    Verifier is an expert in verifiying if user goals are met.
+                    Verifier analyzes chat and responds with TERMINATE if the goal is met.
+                    Verifier can consider the goal met if the other agents have prepared the necessary transactions.
+                    """
+                ),
+            llm_config=self.get_llm_config(),
+            human_input_mode="NEVER",
+            code_execution_config=False,
+        )
+
+        autogen_agents = [agent.build_autogen_agent(self, user_proxy, self.get_llm_config()) for agent in self.agents]
+
+        groupchat = GroupChat(
+            agents=autogen_agents + [user_proxy, verifier_agent], 
+            messages=[], 
+            max_round=20,
+            select_speaker_prompt_template = (
+                """
+                Read the above conversation. Then select the next role from {agentlist} to play. Only return the role and NOTHING else.
+                """
+            )
+        )
+        manager = GroupChatManager(groupchat=groupchat, llm_config=self.get_llm_config())
+
+        if silent:
```

Comment:
While this turns everything off, I'd like to find a way for our agents to log important information to the user. If you ask purely informational questions of the agent, it won't print anything informative. For example, "What's the price of ETH?" doesn't print anything when run without `--verbose`.

This is something we can improve in a future PR.